src/com/owncloud/android/ui/activity/FileActivity.java

```diff
@@ -978,26 +524,24 @@ public void onServiceDisconnected(ComponentName component) {
         }
     }
 
-
     @Override
     public FileDownloaderBinder getFileDownloaderBinder() {
         return mDownloaderBinder;
     }
 
-
     @Override
     public FileUploaderBinder getFileUploaderBinder() {
         return mUploaderBinder;
     }
 
-
+    @Override
     public void restart(){
         Intent i = new Intent(this, FileDisplayActivity.class);
         i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
-        i.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
```

Comment:
I'm checkin #1657; I can't reproduce it, but I'm reviewing the implementation of the switch in any case.

The deletion of this flag was in https://github.com/owncloud/android/pull/1559/commits/1ef9521ac83d2133b653e37156d498c32b03b38c , way before the bug #1657 was detected, so it wasn't removed to fix that bug.

I remember some strange behavior browsing the app that could be related to the absence of this flag. I will try to reproduce it and tell you.